04_setup_ironic.sh

```diff
@@ -34,47 +34,19 @@ if  sudo podman pod exists ironic-pod ; then
     sudo podman pod rm ironic-pod -f
 fi
 
-# set password for mariadb
-mariadb_password=$(echo $(date;hostname)|sha256sum |cut -c-20)
-
 # Create pod
 sudo podman pod create -n ironic-pod 
 
-# Start dnsmasq, http, mariadb, and ironic containers using same image
-sudo podman run -d --net host --privileged --name dnsmasq  --pod ironic-pod \
-     -v $IRONIC_DATA_DIR:/shared --entrypoint /bin/rundnsmasq ${IRONIC_IMAGE}
-
+# We start only the httpd and *downloader containers so that we can provide
+# cached images to the bootstrap VM
 sudo podman run -d --net host --privileged --name httpd --pod ironic-pod \
      -v $IRONIC_DATA_DIR:/shared --entrypoint /bin/runhttpd ${IRONIC_IMAGE}
```

Comment:
This appears to be getting used during the deploy (IPA is downlownloading it from this container), I guess the URLs need to change here
https://github.com/openshift-metal3/dev-scripts/blob/d540fd82d62f188a59367afbc3d7444dbf92d5eb/make-bm-worker/main.go#L13
and
https://github.com/openshift-metal3/dev-scripts/blob/d540fd82d62f188a59367afbc3d7444dbf92d5eb/ocp_install_env.sh#L80

Reply:
Ah yeah good point, I'll change it in the install-config for now - ultimately I'd like to remove the image from there and just reference it inside the installer, fixing https://github.com/openshift-metal3/kni-installer/issues/58

Reply:
Ok so it turns out this won't work, because the image URL/checksum passed via the install-config is used for both terraform to deploy the masters, and also to set the Image on the providerSpec for the worker machineset.

So I think we'll have to fix https://github.com/openshift-metal3/kni-installer/issues/58 and remove that install-config interface, such that we can internally decouple the two different image locations.  Digging into that now.

Reply:
latest version in https://github.com/openshift/installer/pull/2079 should resolve this I think

Reply:
This doesn't seem to work, is the idea the bootstrap should get the image from the httpd pod running on the provisioning host? It's currently downloading from the openshift URL, which is very slow from some of our boxes. In fact, on my hardware in one of data centers, I get a lousy 150kbps, which takes longer than 30 min -- and thus the installer timeout -- to download RHCOS. I'm wondering if that's why we haven't seen this pass in CI.

Reply:
Local caching wasn't working because the hardcoded RHCOS image in dev-scripts didn't match the new version in the installer. I pushed an update here, so the newer RHCOS does get downloaded and cached.

It will never work in CI as written, because we don't cache the RHCOS images in CI. This is trickier to solve, but I'll try fixing it once I actually get the rest of the stuff working on my local box.